dali/operators/image/resize/resize_op_impl.h

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_OPERATORS_IMAGE_RESIZE_RESIZE_OP_IMPL_H_
+#define DALI_OPERATORS_IMAGE_RESIZE_RESIZE_OP_IMPL_H_
+
+#ifndef DALI_RESIZE_BASE_CC
+#error This file is a part of resize base implementation and should not be included elsewhere
+#endif
+
+#include <cassert>
+#include <vector>
+#include "dali/kernels/kernel_manager.h"
+#include "dali/kernels/imgproc/resample.h"
+#include "dali/operators/image/resize/resize_base.h"
+
+namespace dali {
+
+using kernels::ResamplingParams;
+using kernels::ResamplingParamsND;
+
+template <typename Backend>
+struct ResizeBase<Backend>::Impl {
+  using input_type =  typename Workspace::template input_t<Backend>::element_type;
```

Comment:
```suggestion
  using InputType =  typename Workspace::template input_t<Backend>::element_type;
```
Would read better IMO